src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java

```diff
@@ -52,6 +51,36 @@ public void processDelete(Message<ResourceChangedData> resourceChangedMessage) {
         logger.trace(String.format("Resource changed message for deleted event of kind %s "
                 + "for company number %s retrieved", payload.getResourceKind(), companyNumber));
 
+        final ApiResponse<CompanyProfile> response =
+                companyProfileService.getCompanyProfile(logContext, companyNumber);
+        logger.trace(String.format("Retrieved company profile for company number %s: %s",
+                companyNumber, response.getData()));
+        handleResponse(HttpStatus.valueOf(response.getStatusCode()), logContext,
+                "Response from GET call to company profile api", logMap, logger);
+        var data = response.getData().getData();
+        var links = data.getLinks();
+
+        if (links != null && links.getInsolvency() == null) {
+            logger.trace(String.format("Company profile with company number %s,"
+                            + " does not contain insolvency links, will not perform patch",
+                    companyNumber));
+            return;
+        }
+
+        data.setHasInsolvencyHistory(false);
```

Comment:
Sorry Sameer. Differently to what we discussed, please do not set the InsolvencyHistory to false. I re-discussed this with Zahid. Please change also the related changes for the tests.

Reply:
code change done